### DISCUSSION TITLE
Bash

### DISCUSSION THREAD

**Discussion by saladuit** [codewars/codewars.com#2714] [General]
Hey, I'm a 42 student based in Amsterdam, and I'm loving your website. I've a question regarding the languages. 
Will you add bash as a language?
It is very useful for writing acceptance tests, and there is also a cool framework for the language.
https://github.com/bats-core/bats-core

I look forward to hear from you!

**Comment by Blind4Basics** [comment]
Hi,

there is powershell already

**Comment by hobovsky** [comment]
Bash is available as one of versions of Shell language.
Just search for Shell kata and check which of them is available in bash, but I believe it's most of them.

**Comment by saladuit** [comment]
Thank you! I've found it. Stupdid that I couldn't think of that😅. I have a follow up question. Is it possible to add another testing framework to code wars? I was looking at [this comparison](https://github.com/dodie/testing-in-bash) and it seems that shellspec is the most up to date unit testing framework for bash.

**Reply by hobovsky** [reply]
You can suggest changes to setup of codewars runner in `codewars/runner` repository: https://github.com/codewars/runner/issues

**Reply by saladuit** [reply]
Thank you!

**Comment by dcrego** [comment]
Hi, even though there are Powershell and Shell language options, the katas for these are not natural problems for shells.
A natural problem for shells are related to **OS** or **networking** for instance.
Even SQL uses a diferent kind of engine, why don't we get a different engine for shells as well?

**Reply by saladuit** [reply]
It's not 100% true, as there are definitely some instances that you need the same set of skills, for example to run some sort of test script.
